web/modules/custom/ingredient_term/ingredient_term.module

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * @file
+ * Contains hook for Ingredient Term module.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\taxonomy\Entity\Term;
+
+/**
+ * Implements hook_entity_presave().
+ */
+function ingredient_term_entity_presave(EntityInterface $entity): void {
+  if ($entity->getEntityTypeId() === 'node' && $entity->getType() === 'article') {
```

Comment:
If `hook_ENTITY_TYPE_presave` is used there won't be a need to check for the entity type, as that is done by Drupal before calling the hook.